docs/compatibility.md

```diff
@@ -64,6 +64,20 @@ is generally equivalent to
 client = httpx.Client(**kwargs)
 ```
 
+The HTTPX equivalent of `prepare_request` of `requests.Session` instance is `build_request` of `httpx.Client`.
```

Comment:
I’m not sure these are strictly equivalent. I remember `.prepare_request` mutates à request object in place, while `.build_request()` is mostly a convenience helper for applying client defaults.

So maybe we can present it like this...

- There’s no need for `prepare_request()` in HTTPX: request objects are ready to be sent across the wire as soon as they are instantiated, e.g. `Request("GET", "https://example.org")`.
- If you need to build a request while applying client defaults, see <link to existing docs on `build_requests` in the advanced docs>.

I guess both pieces of info can be moved to a « Request instantiation » section, together with the contents of request arguments below?

Reply:
True, they are not strictly equivalent. we can only send `PreparedRequest` to `.send()` in `requests`. And having separate section Request instantiation also makes it clear.